### PR TITLE
Moved definitions for destinations into operator.rb in order to make …

### PR DESCRIPTION
**Pull request by outsidePasser**
…additional 1822

variants easier.

Tested changes against existing completed games 59438 and 59584 (1822 base),
60717 and 60770 (1822 plus), 60186 and 60826 (1822 MRS), 52001 and 59955
(1822 NRS)

### CODE REVIEW

File: lib/engine/operator.rb [hunk 1]
```diff
@@ -23,6 +24,8 @@ def init_operator(opts)
       @loans = []
       @color = opts[:color]
       @text_color = opts[:text_color] || '#ffffff'
+      @dest_coordinates = opts[:dest_coordinates]
```
Comment (tobymao) [inline lib/engine/operator.rb:27]: rename all of these to destination_x

File: lib/engine/game/g_1822/step/destination_token.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def available_hex(entity, hex)
           end
 
           def destination(entity)
```
Comment (tobymao) [inline lib/engine/game/g_1822/step/destination_token.rb:52]: you don't need this method anymore

File: lib/engine/game/g_1822/game.rb [hunk 3]
```diff
@@ -1944,21 +1931,22 @@ def setup_companies
         end
 
         def setup_destinations
-          self.class::DESTINATIONS.each do |corp, destination|
-            description = if corp == 'LNWR'
-                            "Gets destination token at #{destination} when floated"
+          @corporations.each do |c|
+            next if c.dest_coordinates.nil?
```
Comment (tobymao) [inline lib/engine/game/g_1822/game.rb:1935]: next unless c.destination_coordinates

File: lib/engine/game/g_1822/game.rb [hunk 4]
```diff
@@ -914,12 +901,12 @@ def init_corporations(stock_market)
 
         def init_hexes(_companies, _corporations)
           hexes = super
-
-          self.class::DESTINATIONS.each do |corp, destination|
-            hexes.find { |h| h.id == destination }.tile.icons << Part::Icon.new("../icons/1822/#{corp}_DEST",
-                                                                                "#{corp}_destination")
+          @corporations.each do |c|
+            unless c.dest_coordinates.nil?
```
Comment (tobymao) [inline lib/engine/game/g_1822/game.rb:905]: if c.destination_coordinates

**Comment by outsidePasser** [comment]
Ok.  Will fix those renames shortly.

**Comment by outsidePasser** [comment]
Renames and other comments should be all set now.

**Review by tobymao — APPROVED** [review]

**Pull request merged**